core/src/test/scala/unit/kafka/server/KafkaApisTest.scala

```diff
@@ -196,6 +196,11 @@ class KafkaApisTest extends Logging {
       }
     }
 
+    when(replicaManager.metadataCache)
```

Comment:
All we need to do is to create `MetadataCache` according to `MetadataVersion` used by testing. for example:
```scala
  @Test
  def shouldThrowUnsupportedVersionExceptionOnHandleAddOffsetToTxnRequestWhenInterBrokerProtocolNotSupported(): Unit = {
    metadataCache = MetadataCache.zkMetadataCache(brokerId, IBP_0_10_2_IV0)
    brokerEpochManager = new ZkBrokerEpochManager(metadataCache, controller, None)
    kafkaApis = createKafkaApis(IBP_0_10_2_IV0)
    assertThrows(classOf[UnsupportedVersionException],
      () => kafkaApis.handleAddOffsetsToTxnRequest(null, RequestLocal.withThreadConfinedCaching))
  }
```